### ISSUE TITLE
GPL-938 Improve Heron mistake recovery - remove coupling between 4 HT plates (C=M, V=2)

### ISSUE THREAD

**Issue by neilsycamore** [sanger/limber#678]
The first thing that happens in the Heron-384 pipeline is up to 4 LTHR Cherrypick plates are chosen to form the submission. Once you've made this choice, you are stuck with it, and to redo it you'd have to fail a plate and make a new submission.

There are at least a couple of reasons why users might want to change their mind about which LTHR Cherrypick plates they want, after this step:
1. If a user accidentally scans the wrong plate, or doesn't scan all of them, when creating the submission in Limber (see original reason for this story in [comment](https://github.com/sanger/limber/issues/678#issuecomment-1029167147))
2. If a user only realises later on (when plate goes onto the robot) that there is something wrong with a plate (e.g. contamination), so they want to swap that plate out for another.

To make it more flexible, we'd like to make a second change to the Heron request graph (first was made here - https://github.com/sanger/sequencescape/issues/3451).

**Acceptance Criteria**
- [ ] Each LTHR Cherrypick plate has its own submission (See comments, has tradeoffs)
- [ ] Pooling at the end of the pipeline still works (currently relies on the 4 plates being together in the submission)
- [ ] User experience (happy path) remains the same
- [ ] If I scan the wrong plate into the submission page, I can make a submission separately on the correct plate's page, and then scan the correct 4 into the quad stamp page, without having to fail any plates or re-do any submissions.

N.B. This story was re-written 3/2/2022, see [comment](https://github.com/sanger/limber/issues/678#issuecomment-1029167147) for original version

**Comment by KatyTaylor** [comment]
Seems like user errors like this could have caused this RT as well - https://rt.sanger.ac.uk/Ticket/Display.html?id=717089

**Comment by KatyTaylor** [comment]
@JamesGlover do you think the best way of doing this would be that improvement to the heron request graph - i.e. make the submission on each plate, not all 4?

**Comment by stevieing** [comment]
@KatyTaylor and @JamesGlover to refine the story.

**Comment by KatyTaylor** [comment]
Original version of this story:

Add feature to reduce risk of users creating submissions with fewer than 4 plates (C=M, V=4)

Relates to https://rt.sanger.ac.uk/Ticket/Display.html?id=714525
To create a LTHR 384 lab users scan in some but not all of the 4 plates, go back and scan another plate however the result of this is that further along in the pipeline the samples split out into different pools according to their submission. An SSR then has to add a new seq request on the final pool tube as a workaround. The label created does not have the plate ID on it, which presents a risk of sample swaps. (further explanation in RT ticket)

James G suggestion
Easiest way would probably be something along the lines of:
1.  [      ]
2. [      ]
3. [      ]
4. [      ]
[x] Permit partial submissions
[Submit]

With the latter checkbox toggling if the barcode fields are flagged as required or not
Or [Submit | \/] with the dropdown allowing for partial submission.

**Comment by JamesGlover** [comment]
I've crossed out the top requirement, as it isn't strictly necessary, and may complicate things if the submissions are built (by the delayed job) at different times.

It occurs to me that this would potentially allow for the submissions to be built upfront for >4 plates. This could simplify the process, and could even be automated if it was decided to drop the 96 well pipeline. `Sanger_tailed_artic_v1_96` has only been used 96 times (ie. a single plate). However this could also increase the risk of duplicate submissions.

**Comment by JamesGlover** [comment]
From my slack musings:

> Added a minor comment. We shouldn’t actually need a submission per plate, although I from a data-clarity perspective I can see why it would be nice. However, I think the UX would be slightly simpler if we had a single submission still, as you’d know when one plate was ‘ready’ all of them would be.
[08:41](https://psd-team.slack.com/archives/D010F8567RU/p1643964094527319)
Also might be worth thinking if we could streamline the process a bit if we’re no longer needing to know which plates go together at the beginning of the process. Might be scope to either entirely automate the submission (If we can assume the 96 well version is exceedingly rare) or at least have a unified ‘reception’ step to handle it in bulk.
[08:42](https://psd-team.slack.com/archives/D010F8567RU/p1643964120977869)
That would be something to float in front of the heron teams though.
[08:43](https://psd-team.slack.com/archives/D010F8567RU/p1643964216853239)
Hmm… actually thinking about it, adding that functionality would be quite simple, as we wouldn’t even need to change limber. Just have the heron factory build a submission per plate. The presenter would then handle it all correctly, and even show it building if they were very quick off the mark.
[08:45](https://psd-team.slack.com/archives/D010F8567RU/p1643964313112809)
If we built a submission per plate, cancelling would be pretty simple if you DID want the 96 well version, and then you could use the limber interface. Alternatively we update the ‘StateChanger’ on the input plate to enable canceling there.
[08:45](https://psd-team.slack.com/archives/D010F8567RU/p1643964339004399)
And it has only just occurred to me that an updated state-changer is the solution to pre-emptive failing/cancelling

TL/DR: Once this is in place, adding automated submissions becomes as easy as adding the submission to the heron plate factory. Per-plate submissions in this case would facilitate  cancelling more easily, but an update state-changer could enable this from within Limber (and worth for multi-plate submissions too)

**Comment by sdjmchattie** [comment]
I asked Anastasia to check this issue is still required.  She needed to check with some plates so I created 4 dummy source plates and then 4 dummy destination plates in UAT.  Anastasia confirmed that she is no longer able to recreate this issue so it is officially deprecated and no longer needed.

No longer required, won't fix.
